plugins/ar_spectral.py

```diff
@@ -932,6 +968,22 @@ def _get_sem_survey(self):
         logging.warning("No SEM survey stream found")
         return None
 
+    def _find_spectrometer(self, detector):
+        """
+        Find a spectrometer which wraps the given detector
+        return (Detector): the spectrometer
+        raise LookupError: if nothing found.
+        """
+        for spec in self.main_app.main_data.spectrometers:

```

Comment:
You could replace main_app.main_data with main_data, as you did earlier for getting self._acqui_tab.